go/libraries/doltcore/sqle/mergeable_indexes_setup_test.go

```diff
@@ -0,0 +1,320 @@
+		t:               di.t,
+		countLookups:    di.countLookups,
+		finalRanges:     di.finalRanges,
+	}, err
+}
+func (di *testMergeableDoltIndex) AscendGreaterOrEqual(keys ...interface{}) (sql.IndexLookup, error) {
+	indexLookup, err := di.doltIndex.AscendGreaterOrEqual(keys...)
+	return &testMergeableIndexLookup{
+		doltIndexLookup: indexLookup.(*doltIndexLookup),
+		t:               di.t,
+		countLookups:    di.countLookups,
+		finalRanges:     di.finalRanges,
+	}, err
+}
+func (di *testMergeableDoltIndex) AscendLessThan(keys ...interface{}) (sql.IndexLookup, error) {
+	indexLookup, err := di.doltIndex.AscendLessThan(keys...)
+	return &testMergeableIndexLookup{
+		doltIndexLookup: indexLookup.(*doltIndexLookup),
+		t:               di.t,
+		countLookups:    di.countLookups,
+		finalRanges:     di.finalRanges,
+	}, err
+}
+func (di *testMergeableDoltIndex) AscendRange(greaterOrEqual, lessThanOrEqual []interface{}) (sql.IndexLookup, error) {
+	indexLookup, err := di.doltIndex.AscendRange(greaterOrEqual, lessThanOrEqual)
+	return &testMergeableIndexLookup{
+		doltIndexLookup: indexLookup.(*doltIndexLookup),
+		t:               di.t,
+		countLookups:    di.countLookups,
+		finalRanges:     di.finalRanges,
+	}, err
+}
+func (di *testMergeableDoltIndex) DescendGreater(keys ...interface{}) (sql.IndexLookup, error) {
+	indexLookup, err := di.doltIndex.DescendGreater(keys...)
+	return &testMergeableIndexLookup{
+		doltIndexLookup: indexLookup.(*doltIndexLookup),
+		t:               di.t,
+		countLookups:    di.countLookups,
+		finalRanges:     di.finalRanges,
+	}, err
+}
+func (di *testMergeableDoltIndex) DescendLessOrEqual(keys ...interface{}) (sql.IndexLookup, error) {
+	indexLookup, err := di.doltIndex.DescendLessOrEqual(keys...)
+	return &testMergeableIndexLookup{
+		doltIndexLookup: indexLookup.(*doltIndexLookup),
+		t:               di.t,
+		countLookups:    di.countLookups,
+		finalRanges:     di.finalRanges,
+	}, err
+}
+func (di *testMergeableDoltIndex) DescendRange(lessOrEqual, greaterOrEqual []interface{}) (sql.IndexLookup, error) {
+	indexLookup, err := di.doltIndex.DescendRange(lessOrEqual, greaterOrEqual)
+	return &testMergeableIndexLookup{
+		doltIndexLookup: indexLookup.(*doltIndexLookup),
+		t:               di.t,
+		countLookups:    di.countLookups,
+		finalRanges:     di.finalRanges,
+	}, err
+}
+
+// Lookup made to test mergeable indexes by intercepting the lookup functions and adding tracking for testing.
+type testMergeableIndexLookup struct {
+	*doltIndexLookup
+	t            *testing.T
+	countLookups func(int)
+	finalRanges  func([]lookup.Range)
+}
+
+func (il *testMergeableIndexLookup) IsMergeable(indexLookup sql.IndexLookup) bool {
+	return il.doltIndexLookup.IsMergeable(indexLookup.(*testMergeableIndexLookup).doltIndexLookup)
+}
+func (il *testMergeableIndexLookup) Intersection(indexLookups ...sql.IndexLookup) (sql.IndexLookup, error) {
+	il.countLookups(len(indexLookups) + 1) // include 1 for the caller
```

Comment:
`countLookups` essentially lets us know that we've unioned and intersected the expected number of lookups. For instance `v1 = 1` doesn't require either of the two, so the `lookupCount` ends up being 0, while `v1 = 1 OR v1 = 2` ends up being 2 since that's two separate lookups.

Reply:
See my comment below. I don't think these things are pulling their weight, and they make the tests more complex than they need to be (and potentially more prone to breakage if we change the algorithm).